galley/pkg/runtime/monitoring/monitoring.go

```diff
@@ -134,6 +191,35 @@ func newView(measure stats.Measure, keys []tag.Key, aggregation *view.Aggregatio
 	}
 }
 
+func getStateTypeConfigKeys() ([]tag.Key, error) {
+	var err error
+	if NamespaceTag, err = tag.NewKey(namespace); err != nil {
+		panic(err)
+	}
+	if NameTag, err = tag.NewKey(name); err != nil {
+		panic(err)
+	}
+	if VersionTag, err = tag.NewKey(version); err != nil {
+		panic(err)
+	}
+
+	return []tag.Key{NamespaceTag, NameTag, VersionTag}, err
+}
+
+func registerNewStateTypeConfigView(collection string) error {
```

Comment:
is this to future-proof old versions of galley for evolving configuration? this seems like a bunch of complication to go through for monitoring. dynamic registration makes me a bit nervous.

Reply:
this helps gets data for all the instances galley has and we don't have to hard code anything specifically...